subsys/nrf_rpc/nrf_rpc_uart.c

```diff
@@ -56,11 +59,24 @@ static void work_handler(struct k_work *work)
 	uint8_t *data;
 	size_t len;
 	int ret;
+	uint16_t crc_received = 0;
+	uint16_t crc_calculated = 0;
+
 	while (!ring_buf_is_empty(&uart_tr->rx_ringbuf)) {
 		len = ring_buf_get_claim(&uart_tr->rx_ringbuf, &data, NRF_RPC_MAX_FRAME_SIZE);
 		for (size_t i = 0; i < len; i++) {
 			if (hdlc_input_byte(uart_tr, data[i]) == 0) {
 				if (uart_tr->hdlc_state == hdlc_state_frame_found) {
+					crc_received = (uint16_t)uart_tr->frame_buffer[uart_tr->frame_len];
+					crc_received |= (uint16_t)uart_tr->frame_buffer[uart_tr->frame_len + 1] << 8;
+					crc_calculated = crc16_ccitt(0xffff, uart_tr->frame_buffer, uart_tr->frame_len);
+					if (crc_calculated != crc_received) {
+						LOG_DBG("Invalid crc");
+						LOG_HEXDUMP_DBG((uint8_t*)&crc_calculated, CRC_SIZE, "CRC calculated");
```

Comment:
`LOG_HEXDUMP` sounds like an overkill for two bytes :). How about a single `LOG_ERR("Invalid CRC, calculated %x received %x`.

Reply:
fixed.